test/common/testlib.py

```diff
@@ -654,6 +662,94 @@ def snapshot(self, title, label=None):
             attach(filename)
             print("Wrote HTML dump to " + filename)
 
+    def assert_pixels(self, selector, key, ignore=[]):
+        """Compare the given element with its reference"""
+
+        if Image and self.pixels_label and self.cdp and self.cdp.valid:
+            rect = self.call_js_func('ph_element_clip', selector)
+
+            @dataclass
+            class Rect:
+                x0: int
+                y0: int
+                x1: int
+                y1: int
+
+            def relative_clip(sel):
+                try:
+                    r = self.call_js_func('ph_element_clip', sel)
+                    return Rect(r['x'] - rect['x'],
+                                r['y'] - rect['y'],
+                                r['x'] - rect['x'] + r['width'],
+                                r['y'] - rect['y'] + r['height'])
+                except Error:
+                    return None
+
+            ignore_rects = list(map(relative_clip, ignore))
+            base = self.pixels_label + "-" + key
+            filename = base + "-pixels.png"
+            ref_filename = os.path.join(TEST_DIR, "reference", filename)
+            ret = self.cdp.invoke("Page.captureScreenshot", clip=rect, no_trace=True)
+            if "data" in ret:
+                png_now = base64.standard_b64decode(ret["data"])
+                png_ref = os.path.exists(ref_filename) and open(ref_filename, "rb").read()
+                if not png_ref:
+                    with open(filename, 'wb') as f:
+                        f.write(png_now)
+                    attach(filename)
+                    print("New pixel test reference " + filename)
+                    self.failed_pixel_tests += 1
+                else:
+                    img_now = Image.open(io.BytesIO(png_now)).convert("RGBA")
+                    img_ref = Image.open(io.BytesIO(png_ref)).convert("RGBA")
+                    img_delta = img_ref.copy()
+
+                    def ignoreable_coord(x, y):
```

Comment:
Most imporant review comment of them all: Is it "ignorable" or "ignoreable"? Or "ignored" :grinning:

Reply:
it's "ignorable" :-)